aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -715,38 +712,16 @@ impl VMChangeSet {
         Ok(())
     }
 
-    fn squash_additional_reads_needing_exchange<K>(
-        reads_needing_exchange: &mut BTreeMap<StateKey, (WriteOp, Arc<MoveTypeLayout>)>,
-        additional_reads_needing_exchange: BTreeMap<StateKey, (WriteOp, Arc<MoveTypeLayout>)>,
-        skip: &BTreeMap<StateKey, K>,
+    fn squash_additional_reads_needing_exchange<U, V>(
```

Comment:
as I mentioned on the other PR - this gets changed drastically in https://github.com/aptos-labs/aptos-core/pull/10879 - can you undo the change to this function, or rebase after that PR lands?